modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -83,6 +88,9 @@ public PagingProviderProducer(PagingProvider<Object, T> delegate,
     this.supportsOAuth = supportsOAuth;
     retryPolicy = (RetryPolicyTemplate) executionContext.getRetryPolicyTemplate().orElseGet(NoRetryPolicyTemplate::new);
     connectionSupplierFactory = createConnectionSupplierFactory();
+    ConfigurationStats configurationStats = config.getStatistics();
+    mutableStats =
```

Comment:
this ternary (or something very similar) is showing up in a lot of places. Extract to `org.mule.runtime.module.extension.internal.util.MuleExtensionUtils`